src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeDefinition.xml

```diff
@@ -194,7 +194,7 @@
   </Node>
 
   <Node Name="Columns" LocLabel="SR.SchemaHierarchy_Columns" BaseClass="ModelBased" Strategy="PopulateParentDetails" NodeType="Column" ChildQuerierTypes="SqlColumn" DisableSort=""/>
-  <Node Name="Keys" LocLabel="SR.SchemaHierarchy_Keys" BaseClass="ModelBased" Strategy="ElementsInRelationship" NodeType="Key" ChildQuerierTypes="SqlIndex;SqlForeignKeyConstraint" >
+  <Node Name="Keys" LocLabel="SR.SchemaHierarchy_Keys" BaseClass="ModelBased" Strategy="ElementsInRelationship" NodeType="Key" ChildQuerierTypes="SqlIndex;SqlForeignKeyConstraint" ValidFor="NotSqlDw">
```

Comment:
DW doesn't have keys? Is this a bug in the SMO implementation or a real thing?

Reply:
If it's a bug, we should raise with their team and track internally...

Reply:
SSMS object explorer doesn't have a keys folder in dw database

---
In reply to: [154439990](https://github.com/Microsoft/sqltoolsservice/pull/557#discussion_r154439990) [](ancestors = 154439990)

Reply:
Yeah I was wondering if that's actually by design. Looking at unsupported features in https://docs.microsoft.com/en-us/azure/sql-data-warehouse/sql-data-warehouse-tables-overview, they aren't supported so you're right. We should probably take a look at that and see if there are any other things we should have filtered for DW too.